namer/namer.cc

```diff
@@ -888,4 +907,13 @@ ast::ParsedFile Namer::run(core::MutableContext ctx, ast::ParsedFile tree) {
     return tree;
 }
 
+ast::ParsedFile Namer::run(core::MutableContext ctx, shared_ptr<NamerCtx> namerCtx, ast::ParsedFile tree) {
```

Comment:
WDYT about having Namer take a `vector<ast::ParsedFile>` so that it doesn't need a `NamerCtx` and the entire state can be part of treeMap?

Reply:
We do some incremental progress tracking and error flushing in pipeline.cc that wouldn't play nicely with that...

https://github.com/sorbet/sorbet/blob/40adde4f0025ddf78148b94201aaa98b87a3c0e5/main/pipeline/pipeline.cc#L309-L324

Reply:
Got this working